JavaFXUserInterface/src/main/java/SystemTypeEditController.java

```diff
@@ -0,0 +1,95 @@
+import javafx.scene.input.MouseEvent;
+
+import java.net.URL;
+import java.util.ResourceBundle;
+
+public class SystemTypeEditController implements Initializable {
+
+    @FXML
+    private Button systemMenuBtn;
+    @FXML
+    private Button systemTypeMenuBtn;
+    @FXML
+    private Button infoSaveBtn;
+    @FXML
+    private Button infoCancelBtn;
+    @FXML
+    private TextField systemTypeName;
+    @FXML
+    private ImageView systemTypeImageView;
+
+    private UIUtilities uiUtilities;
+
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        uiUtilities = new UIUtilities();
+        attachEvents();
+        setImage();
+    }
+
+    /**
+     * Attaches events to elements in the scene.
+     *
+     * @author Najim
+     */
+    public void attachEvents() {
+        // Change scenes to Systems.fxml
+        systemMenuBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent mouseEvent) {
+                uiUtilities.changeScene(mouseEvent, "/Systems");
+            }
+        });
+        //Attach link to systemTypeMenuBtn to go to SystemTypeList.fxml
+        systemTypeMenuBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent mouseEvent) {
+                uiUtilities.changeScene(mouseEvent, "/SystemTypeList");
+            }
+        });
+//        //Attach link to infoEditBtn to go to SystemTypeEdit.fxml
+//        infoSaveBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+//            @Override
+//            public void handle(MouseEvent mouseEvent) {
+//                uiUtilities.changeScene(mouseEvent, "/SystemTypeEdit");
+//            }
+//        });
+        //Attach link to infoEditBtn to go to SystemTypeEdit.fxml
+        infoCancelBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent mouseEvent) {
+                uiUtilities.changeScene(mouseEvent, "/SystemTypeInfo");
+            }
+        });
+//        //Attach function to deleteBtn to delete an asset type
+//        infoDeleteBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+//            @Override
+//            public void handle(MouseEvent mouseEvent) {}
+//        });
+//        //Attach function to deleteBtn to delete an asset type
+//        modelDeleteBtn.setOnMouseClicked(new EventHandler<MouseEvent>() {
+//            @Override
+//            public void handle(MouseEvent mouseEvent) {}
+//        });
+    }
+
+    /**
+     * Changes the Image depending on the System Type.
+     *
+     * @author Najim
+     */
+    public void setImage() {
+        if (systemTypeName.getText().equals("Engine")) {
+            systemTypeImageView.setImage(new Image("imgs/system_type_engine.png"));
+        } else {
+            systemTypeImageView.setImage(new Image("imgs/system_type_engine.png"));
+        }
+    }
```

Comment:
This isn't dynamic. If someone enters a new Asset Type, this won't be able to match an image for that new type.

Reply:
Then we will probably need to implement a feature where the user can choose an image for a system type. Otherwise, I don't see how I could make this dynamic, especially if the user wants to enter a new Asset Type.